src/pytorch_lightning/tuner/auto_gpu_select.py

```diff
@@ -55,16 +55,19 @@ def pick_single_gpu(exclude_gpus: List[int]) -> int:
     for i in range(num_cuda_devices()):
         if i in exclude_gpus:
             continue
+        device = torch.device(f"cuda:{i}")
+        # Check if the device has memory used by another torch process
+        if torch.cuda.memory_usage(device) > 0 and torch.cuda.utilization(device):
```

Comment:
This won't work as expected. I just simulated this by having a first torch process creating a random tensor on gpu. In nvidia-smi you can clearly see the occupied memory.

When starting a second process though and retrieving these variables I get:

```
>>> import torch                                                                                                                                                                      │         
>>> torch.cuda.memory_usage('cuda:0')                                                                                                                                                 
0                                                                                                                                                                                     │         
>>> torch.cuda.memory_reserved('cuda:0')                                                                                                                                              
0
```
indicating that torch only returns the stats for the current process.

Furthermore this has an additional dependency on pynvml.

Reply:
Hmm thanks @justusschock let me confirm this and try to figure out if there is a way we can do this with torch

Reply:
Yeah can confirm this. 

This very odd an not what you'd expect from torch?